apps/api/src/controllers/competition.controller.ts

```diff
@@ -654,6 +655,197 @@ export function makeCompetitionController(services: ServiceRegistry) {
         next(error);
       }
     },
+
+    /**
+     * Join an agent to a competition
+     * @param req AuthenticatedRequest with competitionId and agentId params
+     * @param res Express response object
+     * @param next Express next function
+     */
+    async joinCompetition(
```

Comment:
nre: made a note https://github.com/recallnet/js-recall/issues/451#issuecomment-2917733714 to consider an agent-only route in the future (not needed here) that removes the `.../agents/:agentId` path & param.

Reply:
Good idea